### ISSUE TITLE
How to Reduce the Response

### ISSUE THREAD

**Issue by AravindRohit** [Soulter/hugging-chat-api#108]
from hugchat import hugchat
from hugchat.login import Login
import time

email = 'umadevi741401@gmail.com'
passwd = 'Deviaravind@250723'

# Log in to huggingface and grant authorization to huggingchat
sign = Login(email, passwd)
cookies = sign.login()

# Save cookies to the local directory
cookie_path_dir = "./cookies_snapshot"
sign.saveCookiesToDir(cookie_path_dir)

# Create a ChatBot
chatbot = hugchat.ChatBot(cookies=cookies.get_dict())
chatbot.switch_llm(0)

# Create a new conversation
id = chatbot.new_conversation()
chatbot.change_conversation(id)

def perform_query(query_text):
    # Ask a question to the first model with custom parameters
    start_time = time.time()
    query_result = chatbot.query(
        query_text,
        temperature=0.5,
        top_p=0.8,
        top_k=2,
        retry_count=1,
        stream=False
    )
    query_time = time.time() - start_time

    # print(f"Model switch time: {switch_time:.2f} seconds")
    print(f"Query time: {query_time:.2f} seconds")

    return query_result

# Usage example:
query_text = "What is main difference between list and tuple?"
query_result = perform_query(query_text)
query_text = "What is python?"
query_result1 = perform_query(query_text)
print("Response:")
print(query_result.text)
print(query_result1.text)

**Comment by AravindRohit** [comment]
How to reduce the Response it was generating big 

The main difference between a list and a tuple in Python is that a list is mutable, meaning it can be modified or changed after it's created, while a tuple is immutable, meaning it cannot be modified once it's created.

Here are some key differences between lists and tuples:

1. Mutability: Lists are mutable, which means you can add, remove, or modify elements after the list has been created. Tuples, on the other hand, are immutable, which means you cannot change their contents after they have been created.
2. Syntax: Lists are defined using square brackets [] and items are separated by commas , whereas tuples are defined using parentheses () and items are separated by commas.
3. Indexing: In lists, you can access each element by its index (position) within the list, starting from 0. In tuples, you cannot use indexing to access elements, but you can use the slice notation to extract a subset of the elements.
4. Size: Lists can grow or shrink dynamically as elements are added or removed, while the size of a tuple is fixed and cannot be changed.
5. Methods: Lists have more methods than tuples, such as sort(), reverse(), and extend(), which are not available for tuples.
6. Memory usage: Tuples take less memory space than lists because they don't need to store information about the indices of their elements.
7. Performance: Tuples are generally faster than lists when accessing elements by index, especially for large datasets, since tuples don't need to calculate the index of an element every time it's accessed.
8. Typing: Lists can hold elements of different data types, while tuples can only hold elements of the same data type.
9. Use cases: Lists are typically used for collections that need to be modified frequently, while tuples are often used for small, static sets of data where the order matters, like a pair of coordinates or a set of colors.

In summary, if you need a collection that can be modified frequently and requires frequent insertion, deletion, or rearrangement of elements, use a list. If you need a small, static set of data where the order matters and elements don't need to be changed, use a tuple.

**Comment by ogios** [comment]
maybe you can say:

remember, you can only reply within 50 words for the questions i'm about to give you, here's the question: what are the differences between list and tuple in python?

**Comment by AravindRohit** [comment]
If i ask what is python? or what is machine learning? also it gives response too long how can we  reduce that?

**Comment by AravindRohit** [comment]
In Python, lists and tuples are both data structures that can store a collection of items. However, there are some key differences between them:

1. Mutable vs Immutable: Lists are mutable, which means they can be modified after creation. Tuples, on the other hand, are immutable, meaning once created, their contents cannot be changed.
2. Size: Lists can grow or shrink dynamically as elements are added or removed. Tuples have a fixed size and cannot change.
3. Accessing Elements: Both lists and tuples allow you to access individual elements using indexing (e.g., `my_list[0]`). However, lists also support slicing operations like `my_list[:5]`.
4. Methods: Lists have more methods than tuples, such as `.append()`, `.extend()`, and `.insert()`. Tuples only have a few built-in methods, but they do not support any mutating operations.
5. Memory Allocation: When creating a new list, it is stored in contiguous memory locations. This makes appending and inserting elements at the end of the list efficient. On the other hand, when creating a new tuple, its elements are stored in non-contiguous memory locations, making it less efficient for accessing and modifying elements.
6. Use Cases: Lists are commonly used when the number of elements may vary during runtime, while tuples are often used when the number of elements is known beforehand and does not need to be changed later. Additionally, tuples are useful for storing heterogeneous data types, whereas lists are better suited for homogenous data sets.

**Comment by AravindRohit** [comment]
> In Python, lists and tuples are both data structures that can store a collection of items. However, there are some key differences between them:
> 
> 1. Mutable vs Immutable: Lists are mutable, which means they can be modified after creation. Tuples, on the other hand, are immutable, meaning once created, their contents cannot be changed.
> 2. Size: Lists can grow or shrink dynamically as elements are added or removed. Tuples have a fixed size and cannot change.
> 3. Accessing Elements: Both lists and tuples allow you to access individual elements using indexing (e.g., `my_list[0]`). However, lists also support slicing operations like `my_list[:5]`.
> 4. Methods: Lists have more methods than tuples, such as `.append()`, `.extend()`, and `.insert()`. Tuples only have a few built-in methods, but they do not support any mutating operations.
> 5. Memory Allocation: When creating a new list, it is stored in contiguous memory locations. This makes appending and inserting elements at the end of the list efficient. On the other hand, when creating a new tuple, its elements are stored in non-contiguous memory locations, making it less efficient for accessing and modifying elements.
> 6. Use Cases: Lists are commonly used when the number of elements may vary during runtime, while tuples are often used when the number of elements is known beforehand and does not need to be changed later. Additionally, tuples are useful for storing heterogeneous data types, whereas lists are better suited for homogenous data sets.

I have asked what are the differences between list and tuple in python? this only but it gives many response i need to generate only three sentences with proper meaning

**Comment by AravindRohit** [comment]
import time
from hugchat import hugchat
from hugchat.login import Login

email = 'umadevi741401@gmail.com'
passwd = 'Deviaravind@250723'

# Log in to Hugging Face and grant authorization to Hugging Chat
sign = Login(email, passwd)
cookies = sign.login()

# Save cookies to the local directory
cookie_path_dir = "./cookies_snapshot"
sign.saveCookiesToDir(cookie_path_dir)

# Create a ChatBot
chatbot = hugchat.ChatBot(cookies=cookies.get_dict())
chatbot.switch_llm(2)

# Create a new conversation
id = chatbot.new_conversation()
chatbot.change_conversation(id)


def perform_query(query_text):
    # Ask a question to the current conversation with custom parameters
    start_time = time.time()
    query_result = chatbot.query(
        query_text,
        temperature=0.5,
        top_p=0.5,
        stream=False
    )

    query_time = time.time() - start_time

    print(f"Query time: {query_time:.2f} seconds")

    return query_result  # Return the response


# Usage example:
query_text = "How to create an empty tuple?"
query_result = perform_query(query_text)
print("Response:")
print(query_result.text)
this is my code it is running successfully in Pycharm but in Ubuntu i am getting this error Traceback (most recent call last):
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 124, in new_conversation
    cid = json.loads(resp.text)['conversationId']
KeyError: 'conversationId'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "new_hugchat.py", line 17, in <module>
    chatbot = hugchat.ChatBot(cookies=cookies.get_dict())
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 49, in __init__
    self.current_conversation = self.new_conversation()
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 132, in new_conversation
    raise CreateConversationError(f"Failed to create new conversation. ({err_count})")
hugchat.exceptions.CreateConversationError: Failed to create new conversation. (6)
how to fix this

**Comment by AravindRohit** [comment]
how to run hugchat in ubuntu?

**Comment by Whitelisted1** [comment]
> If i ask what is python? or what is machine learning? also it gives response too long how can we reduce that?

For your first two questions, this is not a general support forum, so it should not be used to ask questions not relating to the package.
For your final question, unfortunately, Huggingface does not allow us to specify the maximum number of tokens the model should generate anymore, so the best way you can shorten the response is by truncating it or asking the bot to keep its responses concise.

**Comment by Whitelisted1** [comment]
> how to run hugchat in ubuntu?

You are able to run hugchat in Ubuntu by following the [install guide](https://github.com/Soulter/hugging-chat-api#installation) and importing it in a python file (see the [usage guide](https://github.com/Soulter/hugging-chat-api#usage)). You are able to test to see if the package has installed properly by initiating the CLI with the command `python -m hugchat.cli -u <hugginface email>` ([more information about the CLI](https://github.com/Soulter/hugging-chat-api#cli)).

**Comment by Whitelisted1** [comment]
> ...
> query_text = "How to create an empty tuple?" query_result = perform_query(query_text) print("Response:") print(query_result.text) this is my code it is running successfully in Pycharm but in Ubuntu i am getting this error Traceback (most recent call last): File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 124, in new_conversation cid = json.loads(resp.text)['conversationId'] KeyError: 'conversationId'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last): File "new_hugchat.py", line 17, in chatbot = hugchat.ChatBot(cookies=cookies.get_dict()) File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 49, in **init** self.current_conversation = self.new_conversation() File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 132, in new_conversation raise CreateConversationError(f"Failed to create new conversation. ({err_count})") hugchat.exceptions.CreateConversationError: Failed to create new conversation. (6) how to fix this

It appears as if you are running an older version of the package (which we can tell from the old error message). Try updating the package with `pip install --upgrade hugchat` and then check to see if the package works then.

**Comment by AravindRohit** [comment]
> > ...
> > query_text = "How to create an empty tuple?" query_result = perform_query(query_text) print("Response:") print(query_result.text) this is my code it is running successfully in Pycharm but in Ubuntu i am getting this error Traceback (most recent call last): File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 124, in new_conversation cid = json.loads(resp.text)['conversationId'] KeyError: 'conversationId'
> > During handling of the above exception, another exception occurred:
> > Traceback (most recent call last): File "new_hugchat.py", line 17, in chatbot = hugchat.ChatBot(cookies=cookies.get_dict()) File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 49, in **init** self.current_conversation = self.new_conversation() File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 132, in new_conversation raise CreateConversationError(f"Failed to create new conversation. ({err_count})") hugchat.exceptions.CreateConversationError: Failed to create new conversation. (6) how to fix this
> 
> It appears as if you are running an older version of the package (which we can tell from the old error message). Try updating the package with `pip install --upgrade hugchat` and then check to see if the package works then.

Successfully installed hugchat-0.3.3
I have installed latest version only 
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 124, in new_conversation
    cid = json.loads(resp.text)['conversationId']
KeyError: 'conversationId'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "new_hugchat.py", line 17, in <module>
    chatbot = hugchat.ChatBot(cookies=cookies.get_dict())
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 49, in __init__
    self.current_conversation = self.new_conversation()
  File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 132, in new_conversation
    raise CreateConversationError(f"Failed to create new conversation. ({err_count})")
hugchat.exceptions.CreateConversationError: Failed to create new conversation. (6)
coming like this in ubuntu only 

import time
from hugchat import hugchat
from hugchat.login import Login

email = 'umadevi741401@gmail.com'
passwd = 'Deviaravind@250723'

# Log in to Hugging Face and grant authorization to Hugging Chat
sign = Login(email, passwd)
cookies = sign.login()

# Save cookies to the local directory
cookie_path_dir = "./cookies_snapshot"
sign.saveCookiesToDir(cookie_path_dir)

# Create a ChatBot
chatbot = hugchat.ChatBot(cookies=cookies.get_dict())
chatbot.switch_llm(2)

# Create a new conversation
id = chatbot.new_conversation()
chatbot.change_conversation(id)


def perform_query(query_text):
    # Ask a question to the current conversation with custom parameters
    start_time = time.time()
    query_result = chatbot.query(
        query_text,
        temperature=0.5,
        top_p=0.5,
        stream=False
    )

    query_time = time.time() - start_time

    print(f"Query time: {query_time:.2f} seconds")

    return query_result  # Return the response


# Usage example:
query_text = "How to create an empty tuple?"
query_result = perform_query(query_text)
print("Response:")
print(query_result.text)
this is my code

**Comment by AravindRohit** [comment]
What Python version we need to use in ubuntu I am using 3.6 in ubuntu machine better we need to install python version

**Comment by Soulter** [comment]
> What Python version we need to use in ubuntu I am using 3.6 in ubuntu machine better we need to install python version

The repo works well at Python3.6+, so you can run it with 3.6

**Comment by AravindRohit** [comment]
Traceback (most recent call last):
File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 124, in new_conversation
cid = json.loads(resp.text)['conversationId']
KeyError: 'conversationId'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "new_hugchat.py", line 17, in
chatbot = hugchat.ChatBot(cookies=cookies.get_dict())
File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 49, in init
self.current_conversation = self.new_conversation()
File "/home/ubuntu/.local/lib/python3.6/site-packages/hugchat/hugchat.py", line 132, in new_conversation
raise CreateConversationError(f"Failed to create new conversation. ({err_count})")
hugchat.exceptions.CreateConversationError: Failed to create new conversation. (6)   

I am getting this error with python 3.6 in Ubuntu machine whether i need to download higher versions

**Comment by AravindRohit** [comment]
In before Version 0.2.4 of hugchat ubuntu responded but after updation of hugchat in ubuntu is failing
